src/replica/Application.cc

```diff
@@ -182,20 +178,27 @@ int Application::run() {
         Configuration::setSchemaUpgradeWaitTimeoutSec(_schemaUpgradeWaitTimeoutSec);
     }
     if (_enableServiceProvider) {
+        _serviceProvider = ServiceProvider::create(_config, _instanceId);
 
-        // Create and then start the provider in its own thread pool before
-        // performing any asynchronous operations via BOOST ASIO.
-        //
+        // Update general configuration parameters.
+        // Note that options specified by a user will have non-empty values.
```

Comment:
Does this imply that options not specified by user have an empty value? Cannot user say `--option=""` which will make an empty value?

Reply:
Yes to both questions. Right now, none of the existing parameters has an empty value. Please, see `json const ConfigurationSchema::_schemaJson` in https://github.com/lsst/qserv/blob/main/src/replica/ConfigurationSchema.cc

Besides, the command line parser that I'm using here supports:
- positional parameters (with a possibility to specify some trailing parameters as optional)
- options (which must have non-empty value)
- flags

Examples:
```
required-1 required-2 [optional-1 [ optional2 ... ]]
--option=val
--flag
```
Configuration parameters in the "general" category are represented by the command-line "options" (as defined above). If I have a logical flag then it would be still represented with an option that would have a numeric value:
```
--logical-option={0|1}
```